### ISSUE TITLE
40.94.*.* is missing

### ISSUE THREAD

**Issue by sfarbota** [MicrosoftDocs/OfficeDocs-Enterprise#544]
It looks like IPs in the 40.94.\*.\* range are coming from Microsoft but are not listed here. I'm guessing they are supposed to be in the \*.protection.outlook.com and \*.mail.protection.outlook.com sections since they seem to have similar traffic patterns as 40.107.\*.\*. Please confirm.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 077406d8-bff1-496e-3ea2-b38558fd9355
* Version Independent ID: 57db3dec-2250-2dee-fd1a-ec6656911798
* Content: [Office 365 URLs and IP address ranges](https://docs.microsoft.com/en-us/office365/enterprise/urls-and-ip-address-ranges#feedback)
* Content Source: [Enterprise/urls-and-ip-address-ranges.md](https://github.com/MicrosoftDocs/OfficeDocs-Enterprise/blob/live/Enterprise/urls-and-ip-address-ranges.md)
* Service: **o365-administration**
* GitHub Login: @JoeDavies-MSFT
* Microsoft Alias: **josephd**

**Comment by mypil** [comment]
@sfarbota  - Thank you for submitting feedback.

I will get this issue over to the Enterprise writing team for investigation.

Thank you for reporting and making the docs better. Much appreciated.

I made a note to request the team to update this when the work is complete.

**Comment by mypil** [comment]
@JoeDavies-MSFT - Can you please share your insights on this issue?

Thank you.

**Comment by JoeDavies-MSFT** [comment]
@sfarbota I have forwarded this request to the product team that manages the list of endpoints for Exchange Online for their consideration in a future update. Thanks for letting us know.

**Comment by sfarbota** [comment]
Thanks, @JoeDavies-MSFT 

Where would be the best place to ask for clarification about the traffic I am seeing from this IP? I regularly seeing significant traffic from these IPs to the following non-existent pages on our websites but am not sure exactly why:

- /dgvybxmtyw
- /lookup/rxh0zxjuyw
- /about/ynvzaw5lc3 (or just /ynvzaw5lc3)

**Comment by andyhardingmando** [comment]
@sfarbota - did you find a resolution to this or where the links to non-existent pages are coming from?

**Comment by sfarbota** [comment]
Yes - they are coming from ATP. Here are some relevant snippets from a Microsoft O365 Support rep:

- I have confirmed that these URLs are the subparts of the ATP services which is used in rewriting the URLs and executing the attachments under the safe attachments. 
- It is not the random URLs but these are part of the ATP execution process which gets executed whenever a safe link/ safe attachment process is performed on an email to redirect URL/ to check the attachment
- This is the functionality with the all ATP implementations. This is something rarely highlighted by anyone that is why it does not have an online documentation and this is an Internal process because of which these URLs are accessed
- Yes, It would be safe to filter out in your analytics report

**Comment by mikeboothdesign** [comment]
@sfarbota Thanks for this. We've also seen these numbers skewing our web reporting and quite possibly causing an issue with our firewall.

It also seems to hit these URLs for us:

- /terms/cHJpdmFjeS
- /reviews

As someone non-technical in this regard, could you help me to understand: what exactly is this requesting from the website? If it is re-writing, why does it hit the live website URLs?

**Comment by sfarbota** [comment]
Unfortunately I have no idea why it is necessary for MS to request a random, non-existent URL from a website while checking an email link/attachment. The above statements are the only insights I was able to glean from their support. We ended up just filtering out the ATP IP ranges from our analytics reports and moving on, but I agree it's a bit frustrating to have to just live with this seemingly unnecessary load on our servers.
